storage.rules

```diff
@@ -2,7 +2,8 @@ rules_version = '2';
 service firebase.storage {
   match /b/{bucket}/o {
     match /{allPaths=**} {
-      allow read, write: if false;
+      // Allow access by all users
+      allow read, write;
```

Comment:
That sounds a bit dangerous 😆 We have several rules defined for firestore. Let's sync on how we can ensure that only a user (and a global admin) can only download his/her certificate.

Reply:
Yes :-) Can you help me with that? Here I am in deep waters @andrashee 🤣

Reply:
I'm also not an expert by all means. My approach would have been:
- serve the `donation-certificates` publicly. I think for our use case, where we just want to send someone the URL retrieved through `getDownloadURL` per email and where that link is directly downloadable (without going through a login), those paths need to be publicly readable.
- to avoid any unwanted leakage in the future, I would leave the remaining paths private?

The admin storage api (when it's execute within a firebase function, the private key is automatically populated) can circumvent those roles. For that, it's important that you load it like this:

```
const { getStorage } = require('firebase-admin/storage');
```

See https://firebase.google.com/docs/storage/admin/start

We do the same for the firestore part. Frontend parts (e.g. the admin react interface or the website) are going through the public sdk and need to follow the security rules. In the functions we are using the admin sdk which can do everything.

So something like this (I don't know the exact syntax)?
```
service firebase.storage {
   match /b/{bucket}/o {
     /donation-certificates/{multi_path=**} {
         allow read;
     }
     match /{allPaths=**} {
       allow read, write: if false;
     }
   }
 }
```

@mkue wdyt?

Reply:
Update: when we use the admin storage linked here https://github.com/socialincome-san/public/pull/161, we can leave this rules to `allow read, write: if false;`. I think this is much better, otherwise everyone could have written to our public storage.